scripts/setup_ubuntu.sh

```diff
@@ -150,3 +150,8 @@ fi
 # --no-binary is required to cause the pip package to use the debian package's native binaries.
 # https://lists.apache.org/thread.html/r4d2e768c330b6545649e066a1d9d1846ca7a3ea1d97e265205211166%40%3Cdev.arrow.apache.org%3E
 PYARROW_WITH_PARQUET=1 run_as_original_user pip3 install --no-binary pyarrow 'pyarrow>=4.0,<5.0.0a0'
+
+# Install CUDA
+curl https://developer.download.nvidia.com/compute/cuda/11.4.2/local_installers/cuda_11.4.2_470.57.02_linux.run \
+  --output /tmp/cuda_11.4.2_470.57.02_linux.run
+sh /tmp/cuda_11.4.2_470.57.02_linux.run --silent --toolkit
```

Comment:
I don't love installing CUDA this way. The CUDA installer does such weird things to the system it's used on. Yuck. But I'm also not sure of what a better way to would be. (Other than using the conda packages in conda, but we are not using this in conda.)

Could we at least make the CUDA install optional? Then we could only turn it on in a few runs to reduce the time spent installing it and number of jobs affected by a failure in this install.

Reply:
I can make the CUDA installation an option to `setup_ubuntu.sh`, no problem. But we have to test that every compiler works with CUDA in every configuration (Debug/Release/Sanitizers for `clang`), because a CUDA incompatibility can pop up in any of them, while not in some other. Though we can disable Open repo CUDA checks for the combinations that are not built in the Enterprise repo CI with CUDA, i.e. `g++-9`, `clang++-10`, `clang++-12`/Debug&Release (only Sanitizers are currently built for `clang++-12`). Please, let me know what you think.

Reply:
By the way, I'm installing here CUDA without a driver, so it should be quite harmless to the destination system.

Reply:
This is true. I'm not sure where to set this. So I'm fine with whatever for the moment, we can tweak it later if needed.